### DISCUSSION TITLE
Do we need to provide scripts for hot updates in prometheus?

### DISCUSSION THREAD

**Discussion by luoluoyuyu** [apache/streampipes#1958] [Dev]
Prometheus provides a hot update mechanism that does not require stopping the service to reload the configuration, do we need to provide scripts for prometheus to update the token on a regular basis?

**Comment by dominikriemer** [comment]
Hi @luoluoyuyu, we are talking about the JWT token for API access, right?
There is a `token/renew` endpoint in the `Authentication` REST interface which we could use.
Would this be the endpoint the script would call? 

Does anyone know if this would be a best practice or is there something we can improve in the core to better handle tokens for such services?

**Reply by luoluoyuyu** [reply]
hi @dominikriemer  
I've provided Prometheus with a way to update the token_file on a regular basis, what do you think of this solution. https://github.com/luoluoyuyu/prometheus-token-refresher
